### PR TITLE
[CALCITE-3676] VolcanoPlanner. dumpGraphviz should handle exception

### PR DESCRIPTION
**Pull request by XuQianJin-Stars**
As illustrated in [CALCITE-3676](https://issues.apache.org/jira/browse/CALCITE-3676)

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java [hunk 1]
```diff
@@ -1117,22 +1117,27 @@ public void setImportance(RelNode rel, double importance) {
...
+        pw.println("Sets:");
+        dumpSets(pw);
+      }
+      if (CalciteSystemProperty.DUMP_GRAPHVIZ.value()) {
+        pw.println();
+        pw.println("Graphviz:");
+        dumpGraphviz(pw);
+      }
+    } catch (Exception ex) {
+      throw new AssertionError("Dumps the internal state ", ex);
```
Comment (hsyuan) [inline core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java]: If we change this way, we won't see CannotPlanException

Comment (hsyuan) [reply core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java]: you shouldn't throw a exception or error here.

Comment (chunweilei) [reply core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java]: +1

**Comment by vlsi** [comment]
-1: the change ignores the exception, so it would be hard to analyze failures in production

File: core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java [hunk 2]
```diff
@@ -1124,15 +1124,22 @@ public void dump(PrintWriter pw) {
...
+        pw.println();
+        pw.println("Graphviz:");
+        dumpGraphviz(pw);
+      }
+    } catch (Exception | AssertionError ex) {
+      pw.print("When Dumping the internal state :");
       pw.println();
-      pw.println("Graphviz:");
-      dumpGraphviz(pw);
+      pw.print(ex);
```
Comment (hsyuan) [inline core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java]: Can we change to
```
pw.println("Error when dumping plan state: \n" + e);
```

**Pull request closed without merging**